File: CSharp/Core/LogViewer.Core/Logging/DataStoreLoggerConfiguration.cs
using System.Drawing;
using Microsoft.Extensions.Logging;

namespace LogViewer.Core;

public class DataStoreLoggerConfiguration
{
    #region Properties
    
    public EventId EventId { get; set; }

    public Dictionary<LogLevel, LogEntryColor> Colors { get; } = new()
    {
        [LogLevel.Trace] = new() { Foreground = Color.DarkGray },
        [LogLevel.Debug] = new() { Foreground = Color.Gray },
        [LogLevel.Information] = new(),
        [LogLevel.Warning] = new() { Foreground = Color.Orange},
        [LogLevel.Error] = new() { Foreground = Color.White, Background = Color.OrangeRed },
        [LogLevel.Critical] = new() { Foreground=Color.White, Background = Color.Red },
        [LogLevel.None] = new(),
    };

    #endregion
}

File: VB/Core/LogViewerVB.Core/Logging/DataStoreLoggerConfiguration.vb
Imports System.Drawing
Imports Microsoft.Extensions.Logging

Public Class DataStoreLoggerConfiguration

#Region "Properties"

    Public Property EventId As EventId

    Public Property Colors As Dictionary(Of LogLevel, LogEntryColor) = New Dictionary(Of LogLevel, LogEntryColor) From
    {
        {LogLevel.Trace, New LogEntryColor() With {.Foreground = Color.DarkGray}},
        {LogLevel.Debug, New LogEntryColor() With {.Foreground = Color.Gray}},
        {LogLevel.Information, New LogEntryColor()},
        {LogLevel.Warning, New LogEntryColor() With {.Foreground = Color.Orange}},
        {LogLevel.Error, New LogEntryColor() With {.Foreground = Color.White, .Background = Color.OrangeRed}},
        {LogLevel.Critical, New LogEntryColor() With {.Foreground = Color.White, .Background = Color.Red}},
        {LogLevel.None, New LogEntryColor()}
    }

#End Region

End Class